### PR TITLE
Refactor `CGenerator.xtend`

### PR DESCRIPTION
**Pull request by housengw**
As part of the effort to port `CGenerator.xtend` to java, I have refactored some of it into separate files. 

This PR has reached a good stopping point and is ready to merge.

### CODE REVIEW

File: org.lflang/src/org/lflang/generator/DockerComposeGenerator.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package org.lflang.generator;
+
+import java.io.File;
+import java.io.IOException;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+
+public class DockerComposeGenerator {
```
Comment (edwardalee) [inline org.lflang/src/org/lflang/generator/DockerComposeGenerator.java:7]: This should have a class comment with a short description of the class and author information.

File: org.lflang/src/org/lflang/generator/c/CActionGenerator.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import org.lflang.generator.ActionInstance;
+import org.lflang.generator.CodeBuilder;
+import org.lflang.generator.GeneratorBase;
+import org.lflang.generator.ReactorInstance;
+import org.lflang.lf.Action;
+import org.lflang.lf.Reactor;
+import org.lflang.lf.ReactorDecl;
+
+
+public class CActionGenerator {
```
Comment (edwardalee) [inline org.lflang/src/org/lflang/generator/c/CActionGenerator.java:17]: This should have a class comment with a short description of the class and author information.

File: org.lflang/src/org/lflang/generator/c/CCoreFilesUtils.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
+package org.lflang.generator.c;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.lflang.TargetProperty.SchedulerOption;
+
+public class CCoreFilesUtils {
```
Comment (edwardalee) [inline org.lflang/src/org/lflang/generator/c/CCoreFilesUtils.java:7]: This should have a class comment with a short description of the class and author information.

**Review by edwardalee — APPROVED** [review]
This looks like a great refactoring!  One issue throughout is this:

Each file should have a class comment with a short description of the class and author information.

File: org.lflang/src/org/lflang/generator/c/CModesGenerator.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package org.lflang.generator.c;
+
+import java.util.List;
+
+import org.lflang.ASTUtils;
+import org.lflang.generator.CodeBuilder;
+import org.lflang.lf.Mode;
+import org.lflang.lf.Reactor;
+
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/CModesGenerator.java:9]: Comment and authors missing

File: org.lflang/src/org/lflang/generator/c/CNetworkGenerator.java [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+import org.lflang.ASTUtils;
+import org.lflang.InferredType;
+import org.lflang.TimeValue;
+import org.lflang.TargetProperty.CoordinationType;
+import org.lflang.federated.serialization.FedROS2CPPSerialization;
+import org.lflang.federated.serialization.SupportedSerializers;
+import org.lflang.generator.CodeBuilder;
+import org.lflang.generator.GeneratorBase;
+import org.lflang.generator.ReactionInstance;
+
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/CNetworkGenerator.java:21]: Comment and authors missing

File: org.lflang/src/org/lflang/generator/c/CParameterGenerator.java [hunk 6]
```diff
@@ -4,8 +4,11 @@
 import java.util.List;
 import org.lflang.generator.ParameterInstance;
 import org.lflang.ASTUtils;
+import org.lflang.generator.CodeBuilder;
 import org.lflang.generator.GeneratorBase;
 import org.lflang.lf.Assignment;
+import org.lflang.lf.Parameter;
+import org.lflang.lf.Reactor;
 import org.lflang.lf.Value;
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/CParameterGenerator.java:13]: Comment and authors missing.

File: org.lflang/src/org/lflang/generator/c/CPortGenerator.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
+package org.lflang.generator.c;
+
+import org.lflang.ASTUtils;
+import org.lflang.generator.CodeBuilder;
+import org.lflang.lf.Input;
+import org.lflang.lf.Output;
+import org.lflang.lf.Reactor;
+import org.lflang.lf.ReactorDecl;
+import static org.lflang.generator.c.CGenerator.variableStructType;
+
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/CPortGenerator.java:10]: Comment and authors missing.

File: org.lflang/src/org/lflang/generator/c/CTimerGenerator.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package org.lflang.generator.c;
+
+import java.util.List;
+import org.lflang.generator.GeneratorBase;
+import org.lflang.generator.TimerInstance;
+
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/CTimerGenerator.java:6]: Comment and authors missing.

File: org.lflang/src/org/lflang/generator/c/CTracingGenerator.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.lflang.federated.FederateInstance;
+import org.lflang.generator.ActionInstance;
+import org.lflang.generator.ReactorInstance;
+import org.lflang.generator.TimerInstance;
+import static org.lflang.util.StringUtil.addDoubleQuotes;
+
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/CTracingGenerator.java:11]: Comment and authors missing.

File: org.lflang/src/org/lflang/generator/c/InteractingContainedReactors.java [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+import org.lflang.federated.FederateInstance;
+import org.lflang.lf.Input;
+import org.lflang.lf.Instantiation;
+import org.lflang.lf.Output;
+import org.lflang.lf.Port;
+import org.lflang.lf.Reaction;
+import org.lflang.lf.Reactor;
+import org.lflang.lf.TriggerRef;
+import org.lflang.lf.VarRef;
+
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/InteractingContainedReactors.java:19]: Comment and authors missing.

File: org.lflang/src/org/lflang/generator/python/PythonPreambleGenerator.java [hunk 11]
```diff
@@ -1,8 +1,15 @@
...
 import java.util.ArrayList;
 import java.util.List;
 import org.lflang.ASTUtils;
+import org.lflang.TargetConfig;
+import org.lflang.TimeValue;
+import org.lflang.TargetConfig.TracingOptions;
+import org.lflang.TargetProperty.CoordinationType;
+import org.lflang.generator.CodeBuilder;
+import org.lflang.generator.c.CPreambleGenerator;
 import org.lflang.lf.Preamble;
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/python/PythonPreambleGenerator.java:14]: Comment and authors missing.

**Review by lhstrh — APPROVED** [review]
I agree with @edwardalee; this looks great! I've flagged the locations where class comments and author info are missing. If you could correct those, that'd be great.

File: org.lflang/src/org/lflang/generator/c/CModesGenerator.java [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * Generates C code to support modal model.
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/CModesGenerator.java:11]:
```suggestion
 * Generates C code to support modal models.
```

**Review by lhstrh — APPROVED** [review]

**Pull request merged**